### PR TITLE
TDP-4001 Modify Conferences via api

### PR DESCRIPTION
**Pull request by vsmith-bw**
## For the Committer

⚠️ Ensure that for this repo (**bandwidth.github.io**) that the pull request change is opened against the branch `stop-gap-v2`

### CODE REVIEW

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://bw-dev-tdp-4001.s3-website-us-east-1.amazonaws.com/

File: voice/methods/conferences/postConferencesConferenceId.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ Bandwidth's Voice API leverages Basic Authentication with your Dashboard API Cre
 | Parameter       | Description                                                                     | Mandatory |
 |:----------------|:--------------------------------------------------------------------------------|:----------|
 | status          | (optional) Setting the conference status to `completed` ends the conference.    | No        |
+| redirectUrl     | (optional) The URL to send the Redirect event which will provide new BXML. Not allowed if `state` is `completed`, but required if `state` is `active`    | No        |
```
Comment (n-miles) [inline voice/methods/conferences/postConferencesConferenceId.md]: This should have a link to the event page

File: voice/bxml/callbacks/conferenceRedirect.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+| conferenceId | The ID of the new conference that was created. |
+| name         | The custom name used to reference this conference. This the name that you included inside the body of the [`<Conference>`](../verbs/conference.md) tag. |
+
+{% common %}
+
+#### Example: Conference redirect
+
+```
+
+```
```
Comment (n-miles) [inline voice/bxml/callbacks/conferenceRedirect.md]: What

File: voice/bxml/callbacks/conferenceRedirect.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{% method %}
+## Conference Redirect Event
+The Conference Redirect event is fired whenever an existing conference is modified via a POST request made to the [/conferences/{conferenceId}](../../methods/conferences/postConferencesConferenceId.md) endpoint
+The response may be either empty or a BXML. Only `<PlayAudio>` and `<SpeakSentence>` are supported, which will be heard by members that enter the conference while still playing.
+
+### Expected response
```
Comment (n-miles) [inline voice/bxml/callbacks/conferenceRedirect.md]: Can you base this branch on my branch? This text has changed

File: voice/bxml/callbacks/conferenceRedirect.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+```http
+HTTP/1.1 204
+```
+
+```http
+HTTP/1.1 200
+Content-Type: application/xml; charset=utf-8
+
+<Response>
+    <!-- <PlayAudio> or <SpeakSentence> BXML verbs to play in the conference -->
```
Comment (n-miles) [inline voice/bxml/callbacks/conferenceRedirect.md]: This has also changed

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://bw-dev-tdp-4001.s3-website-us-east-1.amazonaws.com/

File: voice/about.md [hunk 5]
```diff
@@ -49,12 +49,14 @@
 | [Conference Created](bxml/callbacks/conferenceCreated.md)        | Bandwidth API sends this when a new conference is created.                                                                                                                                      |
 | [Conference Member Join](bxml/callbacks/conferenceMemberJoin.md) | Bandwidth API sends this when a new member joins a conference.                                                                                                                                  |
 | [Conference Member Exit](bxml/callbacks/conferenceMemberExit.md) | Bandwidth API sends this when a member leaves a conference.                                                                                                                                     |
+| [Conference Redirect](bxml/callbacks/conferenceRedirect.md)      | Bandwidth API sends this to the application when a POST request is made to the [/conferences/{conferenceId}](methods/conferences/postConferencesConferenceId.md) endpoint requesting to modify a conference |
```
Comment (CatexB) [inline voice/about.md]:
```suggestion
| [Conference Redirect](bxml/callbacks/conferenceRedirect.md)      | Bandwidth API sends this to the application when a POST request is made to the [/conferences/{conferenceId}](methods/conferences/postConferencesConferenceId.md) endpoint requesting to modify a conference. |
```

File: voice/bxml/about.md [hunk 6]
```diff
@@ -65,12 +65,14 @@ BXML callbacks are HTTP messages that are sent to your application server to not
 | [Conference Created](callbacks/conferenceCreated.md)        | Bandwidth API sends this when a new conference is created.                                                                                                                                 |
 | [Conference Member Join](callbacks/conferenceMemberJoin.md) | Bandwidth API sends this when a new member joins a conference.                                                                                                                             |
 | [Conference Member Exit](callbacks/conferenceMemberExit.md) | Bandwidth API sends this when a member leaves a conference.                                                                                                                                |
+| [Conference Redirect](callbacks/conferenceRedirect.md)      | Bandwidth API sends this to the application when a POST request is made to the [/conferences/{conferenceId}](../methods/conferences/postConferencesConferenceId.md) endpoint requesting to modify a conference |
```
Comment (CatexB) [inline voice/bxml/about.md]:
```suggestion
| [Conference Redirect](callbacks/conferenceRedirect.md)      | Bandwidth API sends this to the application when a POST request is made to the [/conferences/{conferenceId}](../methods/conferences/postConferencesConferenceId.md) endpoint requesting to modify a conference. |
```

File: voice/bxml/callbacks/about.md [hunk 7]
```diff
@@ -28,6 +28,7 @@ We use a 15s request timeout when trying to deliver all callbacks, retrying once
 | [Conference Created](conferenceCreated.md)        | Bandwidth API sends this when a new conference is created.                                                                                                                                                   |
 | [Conference Member Join](conferenceMemberJoin.md) | Bandwidth API sends this when a new member joins a conference.                                                                                                                                               |
 | [Conference Member Exit](conferenceMemberExit.md) | Bandwidth API sends this when a member leaves a conference.                                                                                                                                                  |
+| [Conference Redirect](conferenceRedirect.md)      | Bandwidth API sends this to the application when a POST request is made to the [/conferences/{conferenceId}](../../methods/conferences/postConferencesConferenceId.md) endpoint requesting to modify a conference |
```
Comment (CatexB) [inline voice/bxml/callbacks/about.md]:
```suggestion
| [Conference Redirect](conferenceRedirect.md)      | Bandwidth API sends this to the application when a POST request is made to the [/conferences/{conferenceId}](../../methods/conferences/postConferencesConferenceId.md) endpoint requesting to modify a conference. |
```

File: voice/bxml/verbs/startRecording.md [hunk 8]
```diff
@@ -1,32 +1,35 @@
...
 | fileFormat                   | (optional) The audio format that the recording will be saved as: `mp3` or `wav`.  Default value is `wav`.                                                                                                              |
-| multiChannel                 | (optional) A boolean value indicating whether or not the recording file should separate each side of the call into its own audio channel. Default value is `false`.                                                    |
+| multiChannel                 | (optional) A boolean value indicating whether or not the recording file should separate each side of the call into its own audio channel. Default value is `false`.<br><br>Note: this option will be silently ignored if set on a conference recording; only single-channel recordings are allowed on conferences |
```
Comment (CatexB) [inline voice/bxml/verbs/startRecording.md]:
```suggestion
| multiChannel                 | (optional) A boolean value indicating whether or not the recording file should separate each side of the call into its own audio channel. Default value is `false`.<br><br>Note: this option will be silently ignored if set on a conference recording; only single-channel recordings are allowed on conferences. |
```

Comment (CatexB) [inline voice/bxml/verbs/startRecording.md:21]:
```suggestion
| multiChannel                 | (optional) A boolean value indicating whether or not the recording file should separate each side of the call into its own audio channel. Default value is `false`.<br><br>Note: this option will be silently ignored if set on a conference recording; only single-channel recordings are allowed on conferences. |
```

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://bw-dev-tdp-4001.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://bw-dev-tdp-4001.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://bw-dev-tdp-4001.s3-website-us-east-1.amazonaws.com/

**Review by CatexB — APPROVED** [review]

File: voice/methods/conferences/postConferencesConferenceId.md [hunk 9]
```diff
@@ -18,6 +18,10 @@ Bandwidth's Voice API leverages Basic Authentication with your Dashboard API Cre
 | Parameter       | Description                                                                     | Mandatory |
 |:----------------|:--------------------------------------------------------------------------------|:----------|
 | status          | (optional) Setting the conference status to `completed` ends the conference.    | No        |
+| redirectUrl     | (optional) The URL to send the [conferenceRedirect](../../bxml/callbacks/conferenceRedirect.md)event which will provide new BXML. Not allowed if `state` is `completed`, but required if `state` is `active`    | No        |
+| redirectMethod  | (optional) The HTTP method to use for the request to `redirectUrl` Not allowed if `state` is `completed`    | No        |
+| username        | (optional) The username to send in the HTTP request to `answerUrl` and `diconnectUrl`    | No        |
```
Comment (n-miles) [inline voice/methods/conferences/postConferencesConferenceId.md]: Wrong URL names

File: voice/methods/conferences/postConferencesConferenceId.md [hunk 10]
```diff
@@ -18,6 +18,10 @@ Bandwidth's Voice API leverages Basic Authentication with your Dashboard API Cre
 | Parameter       | Description                                                                     | Mandatory |
 |:----------------|:--------------------------------------------------------------------------------|:----------|
 | status          | (optional) Setting the conference status to `completed` ends the conference.    | No        |
+| redirectUrl     | (optional) The URL to send the [conferenceRedirect](../../bxml/callbacks/conferenceRedirect.md)event which will provide new BXML. Not allowed if `state` is `completed`, but required if `state` is `active`    | No        |
```
Comment (n-miles) [inline voice/methods/conferences/postConferencesConferenceId.md]:
```suggestion
| redirectUrl     | (optional) The URL to send the [conferenceRedirect](../../bxml/callbacks/conferenceRedirect.md) event which will provide new BXML. Not allowed if `state` is `completed`, but required if `state` is `active`    | No        |
```

File: voice/methods/conferences/postConferencesConferenceId.md (hunk 9 above)
Comment (vsmith-bw) [reply voice/methods/conferences/postConferencesConferenceId.md]: Yikes, copy and paste is too powerful for me

**Review by n-miles — APPROVED** [review]

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://bw-dev-tdp-4001.s3-website-us-east-1.amazonaws.com/

File: voice/bxml/callbacks/conferenceRedirect.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+```http
+HTTP/1.1 204
+```
+
+```http
+HTTP/1.1 200
+Content-Type: application/xml; charset=utf-8
+
+<Response>
+    <!-- BXML verbs to play in the conference -->
```
Comment (marcelohossomi) [inline voice/bxml/callbacks/conferenceRedirect.md]:
```suggestion
    <!-- BXML verbs to execute in the conference -->
```

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://bw-dev-tdp-4001.s3-website-us-east-1.amazonaws.com/

**Review by marcelohossomi — APPROVED** [review]

**Pull request merged**